src/core/labeling/qgspallabeling.cpp

```diff
@@ -2429,8 +2429,11 @@ void QgsPalLayerSettings::registerFeature( const QgsFeature &f, QgsRenderContext
             if ( QgsPalLabeling::geometryRequiresPreparation( ddPoint, context, ct ) )
             {
               ddPoint = QgsPalLabeling::prepareGeometry( ddPoint, context, ct );
-              xPos = static_cast< const QgsPoint * >( ddPoint.constGet() )->x();
-              yPos = static_cast< const QgsPoint * >( ddPoint.constGet() )->y();
+              if ( const QgsPoint *point = qgsgeometry_cast< const QgsPoint * >( ddPoint.constGet() ) )
```

Comment:
I don't know this part of the code, but shouldn't there some handling of the failure (early return, etc.) ? otherwise wrong values of xPos, yPos will be used

Reply:
actually what will happen is that the data defined position will be ignored, and the feature will be treated as having no data-defined label position (so the automatic label positioning will kick in). I'll make it log a message though.

Reply:
Sorry, misread the code -- indeed the position will be incorrect!